core/embed/rust/src/ui/model_tr/trace.rs

```diff
@@ -0,0 +1,19 @@
+use crate::trace::Tracer;
+
+use super::component::ButtonPos;
+
+use heapless::String;
+
+pub trait ButtonTrace {
+    /// Describes what happens when a certain button is triggered.
+    fn get_btn_action(&self, _pos: ButtonPos) -> String<25> {
+        "Default".into()
+    }
+
+    /// Report actions for all three buttons in easy-to-parse format.
+    fn report_btn_actions(&self, t: &mut dyn Tracer) {
```

Comment:
make this a method of Flow or something

Reply:
How would I then call this from all the `input_method`s which need to report their button trace?

Reply:
Removed completely